packages/runtime/src/widgets/PopupMenu.tsx

```diff
@@ -38,99 +51,145 @@ interface PopupMenuStyles {
   borderWidth: string;
 }
 export type PopupMenuProps = {
-  [key: string]: unknown;
   items?: PopupMenuItem[];
   widget?: { [key: string]: unknown };
   onItemSelect?: EnsembleAction;
   showDivider?: boolean | Expression<string>;
+  trigger?: "click" | "hover" | "contextMenu";
+  enabled?: boolean;
```

Comment:
What is the use case for enabled? 

Also I think with item-templates we can deprecate the use of showDivider.

Reply:
when `enabled` is set to false; the `widget` provided for PopupMenu will be disabled.
we'd still need showDivider when `items` is being used (and we can't remove items as they have it in [flutter world](https://github.com/EnsembleUI/ensemble/blob/b53a74da0f99b907ed34cc2ea7519c684c77e7d9/modules/ensemble/lib/widget/popup_menu.dart#L33) too)